### PR TITLE
feat: add normalizr

### PR DESCRIPTION
**Pull request by eduardavercosa**
## Objective and Motivation
- Use `normalizr` to process data before it's stored

### CODE REVIEW

File: frontend/js/actions/getBattleList.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 
 import { BATTLE_LIST } from '../constants';
 
+import * as schema from './schema';
+
 function getBattleList() {
   return (dispatch) =>
     getBattleListPage().then((battlesData) => {
-      return dispatch({ type: BATTLE_LIST, payload: battlesData });
+      const normalizedBattleList = normalize(battlesData, schema.battleList);
```
Comment (filipeximenes) [inline frontend/js/actions/getBattleList.js]: I believe you could also do `normalize(battlesData, [schema.battleSchema])`. You probably don't need `schema.battleList`.

File: frontend/js/pages/BattleDetail.js [hunk 2]
```diff
@@ -25,16 +26,16 @@ const Wrapper = styled.section`
...
-  const { loading } = props.battle;
-  const { error } = props.battle;
+const BattleDetail = (props) => {
   const { id } = useParams();
+  const { battles, loading, error } = props;
+  const { user } = props.user;
   useEffect(() => {
-    props.getCurrentUser();
-    props.getBattle(id);
+    if (battles.length === 0 || battles.battle[id]) {
```
Comment (filipeximenes) [inline frontend/js/pages/BattleDetail.js]: I might be misunderstanding something but I'm not sure what `battles.battle[id]` is doing here, perhaps it should be `!battles.battle[id]`, as in "is battle is not yet in the store, fetch it`?

File: frontend/js/pages/BattleDetail.js [hunk 3]
```diff
@@ -43,33 +44,43 @@ function BattleDetail(props) {
   if (error) {
     return 'Ocurred an error';
   }
-  if (!battle) {
+  if (!user) {
```
Comment (filipeximenes) [inline frontend/js/pages/BattleDetail.js]: Be careful with these binary operations in JS, whenever possible use `lodash` methods as they are more readable and more precise on behavior (isEmpty, isNull, isUndefines, ...)

**Review by filipeximenes — COMMENTED** [review]
I didn't run the code but added a few comments, good job overall! 😃

File: frontend/js/pages/BattleDetail.js (hunk 3 above)
Comment (eduardavercosa) [reply frontend/js/pages/BattleDetail.js]: I'll keep that in mind. Thanks!!!

File: frontend/js/pages/BattleDetail.js (hunk 2 above)
Comment (eduardavercosa) [reply frontend/js/pages/BattleDetail.js]: Yes, my bad... just fixed it :)

**Pull request merged**